ng-dev/pr/config.ts

```diff
@@ -6,16 +6,22 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ConfigValidationError, GithubConfig} from '../../utils/config';
-
-import {GithubApiMergeStrategyConfig} from './strategies/api-merge';
+import {ConfigValidationError, GithubConfig} from '../utils/config';
 
 /**
  * Possible merge methods supported by the Github API.
  * https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-button.
  */
 export type GithubApiMergeMethod = 'merge' | 'squash' | 'rebase';
 
+/** Configuration for the Github API merge strategy. */
+export interface GithubApiMergeStrategyConfig {
+  /** Default method used for merging pull requests */
+  default: GithubApiMergeMethod;
+  /** Labels which specify a different merge method than the default. */
+  labels?: {pattern: string; method: GithubApiMergeMethod}[];
+}
+
 /**
  * Configuration for the merge script with all remote options specified. The
  * default `MergeConfig` has does not require any of these options as defaults
```

Comment:
This is something I'm not 100% sure about. Does it make sense to capture the merge-specific configuration in something as generic as `ng-dev/pr`? are there any benefits you foresee here?

Reply:
I actually think we should go a step further and rename this config to PullRequestConfig, while the primary usage is for merging, we rely on it for other actions in the `pr` directory like the `determine-target-branches` (soon to be `info`).  While much of it is about merging PRs, it really it describing how the tooling interacts with PRs.

Additionally, this aligns with how the rest of the ng-dev tooling has a config per primary subcommand (i.e. `ReleaseConfig`, `CaretakerConfig`, `FormatConfig`)

Reply:
That's fair. I wish some of the interfaces, like for the API merge strategy to stay local to their logic, but I don't feel strongly

Reply:
We can look to move the merge strategy back, I wasn't going to move it initially, but it caused a circular dependency iat the build level.  I will move all the symbols used for the ApiMergeStrategy into the api-merge.ts file

Reply:
Actually I missed a piece before that comment, we can't avoid the circular dependency